### DISCUSSION TITLE
need a clarification ( extension from #60 )

### DISCUSSION THREAD

**Discussion by ckdlskgkdrhd** [naverz/zepeto-studio-global#67] [Q&A]
from what I read in #60 
1. we can't pull this off in the studio just yet.
![zzzxzcccc](https://user-images.githubusercontent.com/110361648/187585175-c23e084f-4a42-4454-9873-b9f52848a718.PNG)

2. what about one of those wearable items lying on the ground, when you interact with it, your character setgesture with that item worn (when you move, it disappears).
I can't provide picture for this one as I don't remember which world I've seen it.

3. also for this one, are these all custom animations?
![캡처](https://user-images.githubusercontent.com/110361648/187632927-eb47aa70-e918-43e0-ad25-30b411389808.PNG)
![dsdsaa](https://user-images.githubusercontent.com/110361648/187632932-e8a30f6a-72d2-4f09-91f9-25c0ef1ca6db.PNG)
![fdsdsasd](https://user-images.githubusercontent.com/110361648/187633037-accb76eb-2eda-4a29-bfe0-35d5720b4024.PNG)
![holding a box](https://user-images.githubusercontent.com/110361648/187633050-268a9619-9c82-4ff0-af56-b63dacfa2f8a.PNG)

**Comment by ckdlskgkdrhd** [comment]
I have a gist of what's going on for the 3rd
just need a clarification on the 2nd.

**Reply by jhslee** [reply]
2. Wearable items & Interaction
This will similarly be difficult to accomplish without the use of the mannequin API. An event will be triggered once you interact with the mannequin, after which you'll be able to play the SetGesture for that character. Then, similar to the event system sample i've set up, in #34 you can cancel the gesture, and unequip the items upon detection of movement. 

3. Animations while holding an object 
Yep! These will all be custom running animations for holding an item! 
## Creating a custom Animation:
A. From the zepeto guides, download the creatorBaseSet_zepeto.zip file. (It should contain the fbx file for you to freely edit)
B. Open the fbx file in your 3d modeling tool (Maya Recommended, even though my example shown is blender 😅) 
![image](https://user-images.githubusercontent.com/105688184/187979435-b4239b56-fd8a-494b-ac31-d2e81c1fffd5.png)
C. After importing the completed fbx to Unity Apply the animation type as "Humanoid"
![image](https://user-images.githubusercontent.com/105688184/187979484-e3e6f10a-4ffc-4b23-8e90-2fd8313eae4d.png)
(Mixamo animation used as an example, because i don't work with animations in maya 🙃)
D. Duplicate and edit the controller file found under: Packages> ZEPETO.Character.Controller>Runtime>Resources
![image](https://user-images.githubusercontent.com/105688184/187979509-327637a9-e691-463b-980d-e7fe165d59fb.png)
E. Apply the controller to the animationController variable in ZepetoPlayers object inspector.
![image](https://user-images.githubusercontent.com/105688184/187979527-63cc05e2-cacb-4350-b36d-01e2a0258c45.png)
F. You can edit the animation controller as freely as you with and script using that controller in your code as needed as well!

 ## Adding an item to the player's hands:
A. Dig through the player object's bone structure to find the hand_L/R object. It should be located somewhere deep underneath the "hip" bone. (May need to do a recursive search during runtime as it cannot be added outside of runtime)
B. Zero out the rotation and local position of the object.
C. Add Offsets to the positioning as needed 
![image](https://user-images.githubusercontent.com/105688184/187979708-872225db-93d0-47b5-8fea-1212fc23caa4.png)

**Reply by ckdlskgkdrhd** [reply]
thanks for the response, @jhslee 
I shall patiently wait for the mannequin api.
